### ISSUE TITLE
Checkout error for registered customer with cache_id_prefix on multi server setup

### ISSUE THREAD

**Issue by david-fuehr** [magento/magento2#21842]
This issue comes from `Magento\Framework\Validator\Factory::_initializeConfigList` where absolute file paths are cached. The actual file content is evaluated using the local file system.

The method is used in `\Magento\Customer\Model\ResourceModel\Address::_validate`, `\Magento\Quote\Model\CustomerManagement::validateAddresses` and `\Magento\Customer\Model\ResourceModel\Customer::_validate` and thus the issue occurs whenever a `customer`, a `customer_address` or quote for a registered customer is saved.

We have a fix for this issue in use for quite some time where we stop caching the name of the files (currently two `validation.xml`s). I will soon open a pull request with these changes.

### Preconditions
We experienced this issue from 2.1.8 and PHP 7.0 upwards to:
1. Magento 2.3-develop
2. PHP 7.2.13

### Steps to reproduce (real life scenario)
1. have a multi server setup with
    1. node _backend_ (absolute instance dir: `/var/www/backend/release/2019-02-28`)
    1. node _frontend-a_ (absolute instance dir: `/var/www/frontend/release/2019-02-28`)
    1. node _frontend-b_ (absolute instance dir: `/var/www/frontend/release/2019-03-01` - autospawned)
    1. node _redis_
    1. share cache for all Magento instances (identical `id_prefix` in cache section of `env.php`)
    1. [node _mysql_, ...]
1. clear cache on backend node
1. right after that save a `customer` or `customer_address` on the backend node (may be a cron job or similar)
1. place an order with a registered customer on one of the _frontend_-nodes

### Expected result
1. The success page is being shown

### Actual result
1. The order can not be placed. An error is displayed:
```
The "/var/www/backend-2019-02-28/app/code/Magento/Eav/etc/validation.xml" file doesn't exist.
```

### Development 
For development purposes the setup can be simulated:
1. set up a regular instance
1. make sure that the `id_prefix` in `env.php` is defined
```
    'cache' => [
        'frontend' => [
            'default' => [
                'id_prefix' => '173_'
            ],
            'page_cache' => [
                'id_prefix' => '173_'
            ]
        ]
    ],
```
3. redis may be used, the issue can also be reproduced using file based caching
1. register a customer
1. place an order using this customer
1. move the instance directory to another path
1. update webserver configuration accordingly and restart
1. try to place an order with the same customer again
    1. _expected:_ success page is being shown
    1. _actual:_ error '[...]Magento/Eav/etc/validation.xml" file doesn't exist.'

### Additional
Related PR: https://github.com/magento/magento2/pull/14411 (2.1 - closed)

**Comment by magento-engcom-team** [comment]
Hi @david-fuehr. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@david-fuehr do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98878`, `MAGETWO-98879` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @david-fuehr. Thank you for your report. 
The issue has been fixed in magento/magento2#21856 by @david-fuehr in 2.3-develop branch
Related commit(s):
 - [11065422a5c9da48898241786848cbf04d2632e3](https://github.com/magento/magento2/commit/11065422a5c9da48898241786848cbf04d2632e3)
 - [eb989e2f5807ef4afcd3172eea9f60795afea823](https://github.com/magento/magento2/commit/eb989e2f5807ef4afcd3172eea9f60795afea823)
 - [7d0d01e327a54b460d334165e7a68fcae7456f87](https://github.com/magento/magento2/commit/7d0d01e327a54b460d334165e7a68fcae7456f87)
 - [e266d9af43a01fce13d98001b335bf69b4219eee](https://github.com/magento/magento2/commit/e266d9af43a01fce13d98001b335bf69b4219eee)
 - [83b34b89b5caffb1b2eef74423090e7f343b0e18](https://github.com/magento/magento2/commit/83b34b89b5caffb1b2eef74423090e7f343b0e18)
 - [f3d4d9611fdbbec3c4eddca917e86e3fae272f0d](https://github.com/magento/magento2/commit/f3d4d9611fdbbec3c4eddca917e86e3fae272f0d)
 - [b77d8d0f6b560224112a736f186c74e5eed8ac42](https://github.com/magento/magento2/commit/b77d8d0f6b560224112a736f186c74e5eed8ac42)
 - [875fa4ce039a177d4f0f597f392a68218cfea007](https://github.com/magento/magento2/commit/875fa4ce039a177d4f0f597f392a68218cfea007)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @david-fuehr. Thank you for your report. 
The issue has been fixed in magento/magento2#22805 by @david-fuehr in 2.2-develop branch
Related commit(s):
 - [64682601ec50fb09231de9ff29e51753d8cfd2ef](https://github.com/magento/magento2/commit/64682601ec50fb09231de9ff29e51753d8cfd2ef)
 - [1db265d90369de1a647c8aead9fc7fc600b7008d](https://github.com/magento/magento2/commit/1db265d90369de1a647c8aead9fc7fc600b7008d)
 - [30f3fe1c23492b57266ed65e062d1f9b9f9c6ae7](https://github.com/magento/magento2/commit/30f3fe1c23492b57266ed65e062d1f9b9f9c6ae7)
 - [18f3a05cfff05472f0c22bdfe615913773dc7ad6](https://github.com/magento/magento2/commit/18f3a05cfff05472f0c22bdfe615913773dc7ad6)
 - [6c27d26778442a5e36dba8b9bf4a56e9b62bc1aa](https://github.com/magento/magento2/commit/6c27d26778442a5e36dba8b9bf4a56e9b62bc1aa)
 - [ac0559e60f0269765eacdd66e4e6586ef36e6f34](https://github.com/magento/magento2/commit/ac0559e60f0269765eacdd66e4e6586ef36e6f34)

The fix will be available with the upcoming 2.2.10 release.
